src/sentry/integrations/github/repository.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from sentry.app import locks
+from sentry.models import Integration, Organization, OrganizationOption
+from sentry.plugins import providers
+from sentry.utils.http import absolute_uri
+from uuid import uuid4
+
+WEBHOOK_EVENTS = ['push', 'pull_request']
+
+
+class GitHubRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'GitHub Apps'
+    auth_provider = 'github'
+    logger = logging.getLogger('sentry.plugins.github')
+
+    def get_config(self, organization):
+        choices = []
+        for i in Integration.objects.filter(organizations=organization, provider='github'):
+            choices.append((i.id, i.name))
+
+        return [
+            {
+                'name': 'installation',
+                'label': 'Github Installation',
+                'type': 'choice',
+                'choices': choices,
+                'initial': choices[0][0],
+                'placeholder': 'i dk yet',
+                'help': 'Enter your repository name, including the owner.',
+                'required': True,
+            },
+            {
+                'name': 'name',
+                'label': 'Repository Name',
+                'type': 'text',
+                'placeholder': 'e.g. getsentry/sentry',
+                'help': 'Enter your repository name, including the owner.',
+                'required': True,
+            }
+        ]
+
+    def validate_config(self, organization, config, actor=None):
+        """
+        ```
+        if config['foo'] and not config['bar']:
+            raise PluginError('You cannot configure foo with bar')
+        return config
+        ```
+        """
+        if config.get('name') and config.get('installation'):
+            # this doesn't work yet, need github client
+            client = self.get_client()
+            try:
+                repo = client.get_repo(config['name'])
+            except Exception as e:
+                self.raise_error(e)
+            else:
+                config['external_id'] = six.text_type(repo['id'])
+        return config
+
+    def get_webhook_secret(self, organization):
```

Comment:
i don't think you'll actually need these webhook methods, will you? because i think we set up all the webhook stuff via apps in the github UI

Reply:
erp yeah can remove all that